### PR TITLE
[FLINK-8079][tests] Stop end-to-end test execution after first failure

### PR DESCRIPTION
**Pull request by zentol**
## What is the purpose of the change

This PR modifies the travis maven script to stop the execution of end-to-end tests once the first failure was detected.

### CODE REVIEW

**Comment by StephanEwen** [comment]
I like the change+1

Minor: Is there a way to write `if code != 0 return` in bash? Might be easier to read...

File: tools/travis_mvn_watchdog.sh [hunk 1]
```diff
@@ -543,35 +543,45 @@ case $TEST in
...
-			printf "Running Shaded Presto S3 end-to-end test\n"
-			printf "==============================================================================\n"
-			FLINK_DIR=build-target CLUSTER_MODE=cluster test-infra/end-to-end-test/test_shaded_presto_s3.sh
-			EXIT_CODE=$(($EXIT_CODE+$?))
+			if [ $EXIT_CODE == 0 ]; then
+				printf "\n==============================================================================\n"
+				printf "Running Wordcount end-to-end test\n"
+				printf "==============================================================================\n"
+				FLINK_DIR=build-target CLUSTER_MODE=cluster test-infra/end-to-end-test/test_batch_wordcount.sh
+				EXIT_CODE=$(($EXIT_CODE+$?))
```
Comment (greghogan) [inline tools/travis_mvn_watchdog.sh]: If `$EXIT_CODE == 0` then why not simply set `EXIT_CODE=$?`?

Comment (zentol) [reply tools/travis_mvn_watchdog.sh]: jup we can simplify things now.

**Comment by zentol** [comment]
@greghogan good point, I'll adjust that part as well to combine the error codes of both checks. We don't have to fail immediately after one fails since nothing is printed if the check succeeds.

**Comment by zentol** [comment]
merging.

**Pull request closed without merging**